omegaconf/_impl.py

```diff
@@ -0,0 +1,24 @@
+from typing import Any
+
+from omegaconf import MISSING, DictConfig, ListConfig, ValueNode
+from omegaconf.errors import InterpolationToMissingValueError
+
+
+def _resolve(cfg: Any) -> Any:
+    if isinstance(cfg, DictConfig):
+        for k in cfg.keys():
+            node = cfg._get_node(k)
+            cfg[k] = _resolve(node)
+
+    if isinstance(cfg, ListConfig):
+        for i in range(len(cfg)):
+            node = cfg._get_node(i)
+            cfg[i] = _resolve(node)
+
+    elif isinstance(cfg, ValueNode):
+        try:
+            cfg = cfg._dereference_node()
```

Comment:
I feel like this won't handle well situations where after dereferencing, `cfg` is a container that may contain interpolations inside it. Am I wrong?

Reply:
I suspect you were wrong, but in any case take a look at the latest version of the code.
there are tests covering this.